src/overlays/actors/ovl_En_Bjt/z_en_bjt.c

```diff
@@ -31,49 +58,408 @@ const ActorInit En_Bjt_InitVars = {
+    Player* player = GET_PLAYER(play);
+    EnBjt* this = THIS;
+    s32 itemAP;
+    s32 scriptBranch = 0;
+
+    switch (this->behaviour) {
+        case TOILET_HAND_BEHAVIOUR_WAIT_FOR_ITEM:
+            switch (Message_GetState(&play->msgCtx)) {
+                case 4:
+                case 5:
+                    if (!Message_ShouldAdvance(play)) {
+                        break;
+                    }
+                case 16:
+                    itemAP = func_80123810(play);
+                    if ((itemAP == EXCH_ITEM_DEED_LAND) || (itemAP == EXCH_ITEM_LETTER_TO_KAFEI) ||
+                        (itemAP == EXCH_ITEM_DEED_SWAMP) || (itemAP == EXCH_ITEM_DEED_MOUNTAIN) ||
+                        (itemAP == EXCH_ITEM_DEED_OCEAN) || (itemAP == EXCH_ITEM_LETTER_MAMA)) {
+                        EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_WAITING_MORPH);
+                        this->playedSfx = false;
+                        this->behaviour++;
+                        scriptBranch = 1; // Right item
+                    } else if (itemAP < 0) {
+                        this->playedSfx = false;
+                        this->behaviour++;
+                        scriptBranch = 3; // Not showing item
+                    } else if (itemAP != 0) {
+                        this->playedSfx = false;
+                        this->behaviour++;
+                        scriptBranch = 2; // Wrong item
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_TAKE_ITEM:
+            if (player->exchangeItemId != EXCH_ITEM_NONE) {
+                EnBjt_TakeItem(player->exchangeItemId);
+                player->exchangeItemId = EXCH_ITEM_NONE;
+            }
+            if (EnBjt_Vanish(this)) {
+                this->timer = 60;
+                this->behaviour++;
+                scriptBranch = 1; // Right item
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_USE_ITEM:
+            if (DECR(this->timer) == 0) {
+                EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_FIST); // change while not visible
+                this->playedSfx = false;
+                this->behaviour++;
+            } else if (this->timer == 10) {
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_TOILET_WATER);
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_REAPPEAR:
+            if (EnBjt_Appear(this)) {
+                this->behaviour++;
+                scriptBranch = 1; // Right item
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_REWARD_GIVEN:
+            EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_OPEN_HAND);
+            this->behaviour++;
+            scriptBranch = 1; // Right item
+            break;
+
+        default:
+            break;
+    }
+
+    return scriptBranch;
+}
+
+// Used in Update
+s32 EnBjt_CheckTalk(EnBjt* this, PlayState* play) {
```

Comment:
Is this comment necessary?

Reply:
I suppose I wanted to make clear that it was not an actionfunction, since it's a bit hard to tell with the way this actor works.

Reply:
I'd still say the comment is unnecessary. You can also tell since it returns an `s32`.